### PR TITLE
Kav/multicloud

### PR DESCRIPTION
**Pull request by Kav91**
* use new s3 backed api for standardized pricing
* get cloud regions directly
* add multicloud support (amazon, azure, google)
* add set cloud regions modal
* add messaging if accounts / uuid is not set correctly
* update docs
* get config directly with getDocument

image
image

### CODE REVIEW

File: dist/metadata.json [hunk 1]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+{"EXPERIMENTAL_external":true,"id":"e30b3b37-db06-439a-b030-cd5bcf7ee323","name":"nr1-cloud-optimize","version":"0.1.7","cliVersion":"1.0.1","sdkVersion":2,"email":"opensource@newrelic.com","artifacts":[{"schema":{"schemaType":"LAUNCHER","id":"e30b3b37-db06-439a-b030-cd5bcf7ee323.cloud-optimize-launcher","description":"Cloud Optimizer","displayName":"Cloud Optimize","rootNerdletId":"cloud-optimize-nerdlet","pkgId":"e30b3b37-db06-439a-b030-cd5bcf7ee323","nerdpackId":"e30b3b37-db06-439a-b030-cd5bcf7ee323","artifactId":"cloud-optimize-launcher"},"js":null,"css":null,"icon":"e30b3b37-db06-439a-b030-cd5bcf7ee323--cloud-optimize-launcher.png"},{"schema":{"schemaType":"NERDLET","id":"e30b3b37-db06-439a-b030-cd5bcf7ee323.cloud-optimize-nerdlet","description":"Cloud Optimizer","displayName":"Cloud Optimize","pkgId":"e30b3b37-db06-439a-b030-cd5bcf7ee323","nerdpackId":"e30b3b37-db06-439a-b030-cd5bcf7ee323","artifactId":"cloud-optimize-nerdlet"},"js":["vendors~e30b3b37-db06-439a-b030-cd5bcf7ee323--vendors.js","e30b3b37-db06-439a-b030-cd5bcf7ee323--cloud-optimize-nerdlet.js"],"css":["e30b3b37-db06-439a-b030-cd5bcf7ee323--styles.css"]}]}
```
Comment (jthurman42) [inline dist/metadata.json]: In the future, I'd git rm and ignore `dist/`

File: nerdlets/cloud-optimize-nerdlet/cloud-optimize.js [hunk 2]
```diff
@@ -67,24 +84,32 @@ export default class CloudOptimize extends React.Component {
...
-            if(configs.length === 1 && configs[0].id == "main"){ // set existing config
-                console.log("loading existing config")
-                await this.setState({config: configs[0].document}) 
-            }else{ // write in default config
-                console.log("writing default config")
-                await writeDocument("cloudOptimizeCfg", "main", this.state.config)
-            }
-            resolve()
+        return new Promise((resolve) => {
+            console.log("fetching user config from nerdstorage")
```
Comment (jthurman42) [inline nerdlets/cloud-optimize-nerdlet/cloud-optimize.js]: should remove debug / console logs

File: nerdlets/cloud-optimize-nerdlet/cloud-optimize.js [hunk 3]
```diff
@@ -130,18 +155,33 @@ export default class CloudOptimize extends React.Component {
...
-            console.log(`fetching aws ec2 pricing: ${region}`)
-            // https://cors.io/?https://a0.p.awsstatic.com/pricing/1.0/ec2/region/${project.awsRegion}/ondemand/linux/index.json
-            // cors hack
-            fetch(`https://yzl85kz129.execute-api.us-east-1.amazonaws.com/dev?url=https://a0.p.awsstatic.com/pricing/1.0/ec2/region/${region}/ondemand/linux/index.json`).then((response)=> {
-                return response.json()
-            }).then((myJson)=>{
-                this.setState({awsPricing: myJson})
+            let cloudPromises = Object.keys(config.cloudData).map((cloud)=>{
+                return new Promise((resolve)=>{
+                    fetch(`https://nr1-cloud-optimize.s3-ap-southeast-2.amazonaws.com/${cloud}/regions.json`).then((response) => {
```
Comment (jthurman42) [inline nerdlets/cloud-optimize-nerdlet/cloud-optimize.js]: Should have a config var for the s3 bucket, so folks can store it themselves or point at their own instance of it when deploying to a new NR account. Is this small enough to store in NerdStore instead?

File: nerdlets/cloud-optimize-nerdlet/components/optimizationCandidates.js [hunk 4]
```diff
@@ -180,11 +180,11 @@ export default class OptimizationCandidates extends React.Component {
                                             <Table.Cell>{instance.numCpu}</Table.Cell>
                                             <Table.Cell>{instance.memGB}</Table.Cell>
                                             <Table.Cell>{instance.instanceType}</Table.Cell>
-                                            <Table.Cell>{(instance.awsPrice1 * 720).toFixed(2).replace(/\d(?=(\d{3})+\.)/g, '$&,')}</Table.Cell>
+                                            <Table.Cell>{(instance.instancePrice1 * 720).toFixed(2).replace(/\d(?=(\d{3})+\.)/g, '$&,')}</Table.Cell>
```
Comment (jthurman42) [inline nerdlets/cloud-optimize-nerdlet/components/optimizationCandidates.js]: `720` should be a constant

File: nerdlets/cloud-optimize-nerdlet/cloud-optimize.js (hunk 3 above)
Comment (Kav91) [reply nerdlets/cloud-optimize-nerdlet/cloud-optimize.js]: Moved under #8 as feature enhancement

**Pull request merged**